crates/astria-sequencer/src/accounts/query.rs

```diff
@@ -215,3 +220,84 @@ async fn preprocess_request(
     };
     Ok((address, snapshot, height))
 }
+
+/// Custom deterministic sorting function for `AssetBalance` that sorts by balance in descending
+/// order and then by denom in ascending order. This function should never return `Ordering::Equal`,
+/// as calling `sort_unstable_by` on equal values has a non-deterministic result.
+fn compare_asset_balances(lhs: &AssetBalance, rhs: &AssetBalance) -> std::cmp::Ordering {
+    use std::cmp::Ordering;
+
+    match lhs.balance.cmp(&rhs.balance) {
+        // Denoms should never have the same display name
+        Ordering::Equal => lhs.denom.to_string().cmp(&rhs.denom.to_string()),
```

Comment:
Allocations in a hot loop are nasty (even in edge cases). Can you instead hand-roll a `PartialEq` impl on `Denom`? I think if you `PartialEq for TracePrefixed` and `PartialEq for IbcPrefixed`, and then make the decision to order a denom first by `ibc`, then by `trace-prefixed` (I think that's natural? I let you decide), this should be relatively straight forward.